### PR TITLE
Test labels in `LinkNeighborLoader`

### PR DESCRIPTION
**Pull request by Padarn**
Fixing a small bug in the return type of the labels and and adding a test.

### CODE REVIEW

**Comment by Padarn** [comment]
Faling test here looks irrelevant - flaky test?

File: torch_geometric/loader/link_neighbor_loader.py [hunk 1]
```diff
@@ -31,7 +31,8 @@ def _create_label(self, edge_label_index, edge_label):
             return edge_label_index, edge_label
 
         if edge_label is None:
-            edge_label = torch.ones(num_pos_edges, device=device)
+            edge_label = torch.ones(num_pos_edges,
```
Comment (rusty1s) [inline torch_geometric/loader/link_neighbor_loader.py:34]: Mh, I am actually in favor of keeping it that way since `BCELoss` expects floating-point inputs. Does that work for you?

Comment (Padarn) [reply torch_geometric/loader/link_neighbor_loader.py:34]: I see, in this case do you think the assert below on line 36 should be removed?

Comment (Padarn) [reply torch_geometric/loader/link_neighbor_loader.py:34]: But yes generally fine by me, just wanted to make sure the behaviour was consistent

Comment (rusty1s) [reply torch_geometric/loader/link_neighbor_loader.py:34]: I think the assert is a good check, since in case edge labels are given, we expect that this is a multi-class classification. If it is not, edge labels should not be passed to the loader in the first place.

Comment (Padarn) [reply torch_geometric/loader/link_neighbor_loader.py:34]: right, okay I'm sold on your idea - I'll revert this and update the docstring a bit

File: torch_geometric/loader/link_neighbor_loader.py [hunk 2]
```diff
@@ -215,8 +215,8 @@ class LinkNeighborLoader(torch.utils.data.DataLoader):
             and labels :obj:`1` to :obj:`num_classes` represent the labels of
             positive edges. (default: :obj:`0.0`).
             Note that returned labels are :obj:`torch.float` when it is binary
-            classification (to facilitate easy use with cross entropy loss)
-            and :obj:`torch.long` when is multiclass.
+            classification (to facilitate easy use with :obj:`BCE_loss`)
```
Comment (rusty1s) [inline torch_geometric/loader/link_neighbor_loader.py:218]:
```suggestion
            classification (to facilitate ease-of-use with :meth:`torch.binary_cross_entropy_loss`)
```

File: torch_geometric/loader/link_neighbor_loader.py [hunk 3]
```diff
@@ -215,8 +215,8 @@ class LinkNeighborLoader(torch.utils.data.DataLoader):
...
+            and :obj:`torch.long` when is a multiclass classification problem.
```
Comment (rusty1s) [inline torch_geometric/loader/link_neighbor_loader.py:219]:
```suggestion
            and :obj:`torch.long` when it is a multi-class classification problem.
```

**Review by rusty1s — APPROVED** [review]

**Review by rusty1s — APPROVED** [review]
